### PR TITLE
properly handle string filename including backslashes

### PR DESCRIPTION
**Pull request by konosubakonoakua**
Solve issue [**Illegal back reference** Error when lazyloading from_vscode.lua](https://github.com/L3MON4D3/LuaSnip/issues/441) by replacing `\\` with `/`

### CODE REVIEW

File: lua/luasnip/util/str.lua [hunk 1]
```diff
@@ -20,7 +20,7 @@ end
 
 function M.aupatescape(s)
 	local comma_escaped = s:gsub(",", "\\,")
-    	local backslash_replaced = comma_escaped:gsub("\\", "/")
+  local backslash_replaced = comma_escaped:gsub("\\", "/")
```
Comment (konosubakonoakua) [inline lua/luasnip/util/str.lua:23]: sorry, the tabsize went wrong :(

**Comment by L3MON4D3** [comment]
Oh sorry, could you rebase (or maybe just cherry-pick your commits) onto `origin/aupatescape`? The initial branch included unrelated commits (from #440), but now it should be correct

**Comment by konosubakonoakua** [comment]
> Oh sorry, could you rebase (or maybe just cherry-pick your commits) onto `origin/aupatescape`? The initial branch included unrelated commits (from #440), but now it should be correct

Sorry, im not very familar with GITHUB PullRquest and GIT, can u add the code?

```lua
-- str.lua
function M.aupatescape(s)
	local comma_escaped = s:gsub(",", "\\,")
	local backslash_replaced = comma_escaped:gsub("\\", "/")
	return vim.fn.fnameescape(backslash_replaced)
end

```

**Comment by L3MON4D3** [comment]
Okay, I had another look, I think we cannot just replace `\`, it might occur naturally in an unix-path.
Could you make sure `aupatescape` still works for you with the newest commit?

**Comment by konosubakonoakua** [comment]
> Okay, I had another look, I think we cannot just replace `\`, it might occur naturally in an unix-path. Could you make sure `aupatescape` still works for you with the newest commit?

Yes, the latest commit works well for me. :)

**Comment by L3MON4D3** [comment]
Nice :)

**Pull request closed without merging**